src/transformers/adapters/layer.py

```diff
@@ -449,30 +470,37 @@ def adapters_forward(self, hidden_states, input_tensor):
         )
         if not skip_adapters and (len(set(self.adapters.keys()) & adapter_setup.flatten()) > 0):
             if isinstance(adapter_setup, Stack):
-                hidden_states, _, input_tensor = self.adapter_stack(adapter_setup, hidden_states, input_tensor)
+                hidden_states, _, input_tensor = self.adapter_stack(
+                    adapter_setup, hidden_states, input_tensor, layer_norm
+                )
             elif isinstance(adapter_setup, Fuse):
-                hidden_states = self.adapter_fusion(adapter_setup, hidden_states, input_tensor)
+                hidden_states = self.adapter_fusion(adapter_setup, hidden_states, input_tensor, layer_norm)
             elif isinstance(adapter_setup, Split):
-                hidden_states = self.adapter_split(adapter_setup, hidden_states, input_tensor)
+                hidden_states = self.adapter_split(adapter_setup, hidden_states, input_tensor, layer_norm)
             elif isinstance(adapter_setup, Parallel):
                 # notice that we are overriding input tensor here to keep the same dim as hidden_states for the residual
                 # in case we were blowing up the batch for parallel processing of multiple adapters for the same input
-                hidden_states, input_tensor = self.adapter_parallel(adapter_setup, hidden_states, input_tensor)
+                hidden_states, input_tensor = self.adapter_parallel(
+                    adapter_setup, hidden_states, input_tensor, layer_norm
+                )
             elif isinstance(adapter_setup, BatchSplit):
-                hidden_states = self.adapter_batchsplit(adapter_setup, hidden_states, input_tensor)
+                hidden_states = self.adapter_batchsplit(adapter_setup, hidden_states, input_tensor, layer_norm)
             else:
                 raise ValueError(f"Invalid adapter setup {adapter_setup}")
 
             last_config = self.config.adapters.get(adapter_setup.last())
             if last_config["original_ln_after"]:
-                if self.transformer_layer_norm:
-                    hidden_states = self.transformer_layer_norm(hidden_states + input_tensor)
+                if layer_norm:
+                    hidden_states = layer_norm(hidden_states + input_tensor)
                 else:
                     hidden_states = hidden_states + input_tensor
 
-        elif self.transformer_layer_norm:
-            hidden_states = self.transformer_layer_norm(hidden_states + input_tensor)
+        elif layer_norm:
+            hidden_states = layer_norm(hidden_states + input_tensor)
         else:
             hidden_states = hidden_states + input_tensor
 
         return hidden_states
+
+    def forward(self, hidden_states, input_tensor, layer_norm):
```

Comment:
What is the advantage of passing that with every forward call instead of having a parameter in the class containing the layer norm?

Reply:
The layer norm usually is a child module of the Transformer block module (or part of it as e.g. for BERT). As the `AdapterLayer` usually also is a child module of the same block module, we would have to keep a reference to an attribute of the parent module. As this makes the implementation less clean and poses some issues (cf. #228), I decided to pass the layer norm in the forward pass.

I might have missed some better solution though, please let me know if there are any :)